### ISSUE TITLE
Question: tracking borrow stack for location

### ISSUE THREAD

**Issue by kyku** [rust-lang/miri#1862]
Miri offers a way to track pointer tags, but is it possible to track changes in the borrow stack of  given locations? I'm trying to learn why miri refuses to accept my [code](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=c2286399f8e71228e05f556c33c01d70) . I have the following error:

> trying to reborrow for SharedReadWrite at alloc1752, but parent tag <4401> does not have an appropriate item in the borrow stack

After tracking tag 4401 I can see the following:

>   --> src/main.rs:61:29
>    |
> 61 |     println!("{}", unsafe { &(*p).name });
>    |                             ^^^^^^^^^^ popped tracked tag for item [Unique for <4401>]
> 

Then:

>   --> src/main.rs:61:29
>    |
> 61 |     println!("{}", unsafe { &(*p).name });
>    |                             ^^^^^^^^^^ popped tracked tag for item [Unique for <4401>]

And finally the error code I attached above. But then if there is no appropriate item in the borrow stack, then what items are there?

**Comment by oli-obk** [comment]
The problem is in the creation of the pointer. We should probably show the creation of tags, too, not just when they get popped.

TLDR: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=e630cea44c594a03894952db82201f0a works

Longer version:

the pointer in `ch.parent` is older (lower on the stack) than the pointer in the `Pin<Box<..>>` of `f`. This means when reading it, you pop all younger (higher on the stack) pointers. So when we want to drop the pin box later, the pointer's tag is invalid.

The high-level reason this happens is that you create the `ch.parent` pointer *before* you do `assume_init`.

**Comment by kyku** [comment]
Many thanks for your helpful insight.
Actually, the tag's creation **was** logged:

```
note: tracking was triggered
   --> /home/kyku/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/pin.rs:572:15
    |
572 |         Pin { pointer }
    |               ^^^^^^^ created tag 4377

```
